src/plugins/reporter-bugzilla.c

```diff
@@ -110,15 +110,22 @@ static void set_default_settings(map_string_t *osinfo, map_string_t *settings)
 static void set_settings(struct bugzilla_struct *b, map_string_t *settings)
 {
     const char *environ;
+    g_autofree gpointer login = NULL, pw = NULL, bzurl = NULL;
 
     environ = getenv("Bugzilla_Login");
-    b->b_login = g_strdup(environ ? environ : libreport_get_map_string_item_or_empty(settings, "Login"));
+    if (!environ)
+        login = g_hash_table_lookup(settings, "Login");
+    b->b_login = g_strdup(environ ? environ : (login ? login : ""));
```

Comment:
This (and the related changes) seems a bit convoluted.

What about the following?

```c
login = getenv("Bugzilla_Login");
if (!login)
    login = g_hash_table_lookup(settings, "Login");
b->b_login = g_strdup(login ? : "");
```